client/signup/jetpack-connect/index.jsx

```diff
@@ -234,10 +229,16 @@ class JetpackConnectMain extends Component {
 		if ( ! this.checkProperty( 'isJetpackActive' ) ) {
 			return 'notActiveJetpack';
 		}
-		if ( ! this.checkProperty( 'isJetpackConnected' ) ) {
+		if (
+			! this.checkProperty( 'isJetpackConnected' ) ||
+			( this.checkProperty( 'isJetpackConnected' ) && ! this.checkProperty( 'userOwnsSite' ) )
```

Comment:
Ah yeah, that's a weird one. I'll try to talk it out:

If the site is un-connected, or if the site is connected by someone else, we can set the status to 'notConnectedJetpack'

With 'notConnectedJetpack' status, the current user is able to connect.